python_modules/dagster/dagster/_core/instance/instance.py

```diff
@@ -629,6 +629,19 @@ def get_dynamic_partitions(self, partitions_def_name: str) -> Sequence[str]:
         """
         return self._event_storage.get_dynamic_partitions(partitions_def_name)
 
+    def get_dynamic_partitions_definition_id(
+        self, partitions_def_name: str, calling_context: Optional[DynamicPartitionsStore] = None
+    ) -> str:
+        from dagster._core.definitions.partitions.utils import (
+            generate_partition_key_based_definition_id,
+        )
+
+        calling_context = calling_context or self
```

Comment:
should this be non-optional?

wondering why we don't need to pass it in for `DynamicPartitionsStoreAfterRequests`

Reply:
It's a good point, I think the problem is that DagsterInstance is itself a DynamicPartitionsStore which makes this all weird. But can certainly add the missing callsite...